lib/src/isolate_manager/isolate_manager_io.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:async';
+import 'dart:developer';
+import 'dart:isolate';
+
+import 'package:dash_kit_network/dash_kit_network.dart';
+import 'package:dash_kit_network/src/isolate_manager/isolate_manager.dart';
+
+/// The `IsolateManager` class manages a separate isolate for executing tasks
+/// asynchronously. It implements the `IsolateManagerInterface` interface,
+/// which is not shown here.
+// ignore: prefer-match-file-name
+class IsolateManager extends IsolateManagerInterface {
+  // `resultStream` is a broadcast stream that emits results from tasks
+  // executed in the isolate.
+  late final Stream<Result> resultStream;
+  late final StreamSubscription _errorSubscription;
+  late final StreamSubscription _exitSubscription;
+  late final Isolate _isolate;
+  late final SendPort _sendPort;
+
+  // `_receivePort`, `_errorPort`, and `_exitPort` are `ReceivePort`s that
+  // listen for messages from the isolate.
+  final _receivePort = ReceivePort();
+  final _errorPort = ReceivePort();
+  final _exitPort = ReceivePort();
+  // `_managerReadyCompleter` is a `Completer` that completes when the isolate
+  // is ready to receive tasks.
+  final Completer<void> _managerReadyCompleter = Completer<void>();
+  // `_isolateStatus` keeps track of the current status of the isolate.
+  IsolateStatus _isolateStatus = IsolateStatus.created;
+
+  // `start()` initializes the isolate and sets up listeners for errors and
+  // exit events. It waits for the `sendPort` from the isolate to be received
+  // before setting `_sendPort` and completing `_managerReadyCompleter`.
+  // If `start()` is called multiple times, an assertion error is thrown.
+  @override
+  Future<void> start() async {
+    assert(_isolateStatus == IsolateStatus.created, 'Isolate already started');
+    _isolateStatus = IsolateStatus.initializing;
+    final stream = _receivePort.asBroadcastStream();
+    resultStream = stream.where((message) => message is Result).cast<Result>();
+    _isolate = await Isolate.spawn(
+      _isolateFunction,
+      _receivePort.sendPort,
+      errorsAreFatal: false,
+      onError: _errorPort.sendPort,
+      onExit: _exitPort.sendPort,
+    );
+    _errorSubscription = _errorPort.listen((message) {
+      log('isolate manager error: $message');
+    });
+    _exitSubscription = _exitPort.listen((message) {
+      log('isolate manager exit: $message');
+      _errorSubscription.cancel();
+    });
+
+    final sendPortMessage = await stream.first;
+    _handleMessage(sendPortMessage);
+  }
+
+  /// Stops the isolate by killing it and cancelling error and exit event
+  /// listeners. Throws an assertion error if the isolate is not started.
```

Comment:
let's add the message that we can left subscription as is because when Isolate dies, subscriptions also die

Reply:
We cancel all subscription.